docs/api-markdown-documenter/render-api-documentation.js

```diff
@@ -121,6 +132,55 @@ async function renderApiDocumentation() {
 	console.groupEnd();
 }
 
+/**
+ * Processes documents and generates data required for the nav bar.
+ * @param {Array<Object>} documents - List of documents with apiItem.
+ * @param {ApiItem | undefined} documents.apiItem - The API item that the document is created from. Some documents may not have an apiItem.
+ */
+function buildNavBar(documents) {
+	const validKinds = new Set([
+		ApiItemKind.Class,
+		ApiItemKind.Interface,
+		ApiItemKind.Enum,
+		ApiItemKind.Namespace,
+	]);
+	const { allAPIs, packageMap } = documents.reduce(
+		({ allAPIs, packageMap }, { apiItem }) => {
+			if (apiItem === undefined) {
+				return { allAPIs, packageMap };
+			}
+
+			const { displayName, kind } = apiItem;
+
+			const associatedPackage = apiItem.getAssociatedPackage();
+			const packageName =
+				associatedPackage === undefined
+					? undefined
+					: ApiItemUtilities.getUnscopedPackageName(associatedPackage);
+
+			if (kind === ApiItemKind.Package) {
+				packageMap[displayName] = packageName;
+			} else if (validKinds.has(kind)) {
+				allAPIs[packageName] = allAPIs[packageName] || {};
+				allAPIs[packageName][kind] = allAPIs[packageName][kind] || [];
+				allAPIs[packageName][kind].push(displayName);
+			}
+
+			return { allAPIs, packageMap };
+		},
+		{ allAPIs: {}, packageMap: {} },
+	);
+
+	saveToFile("allAPIs.yaml", allAPIs);
+	saveToFile("packageNameToDisplayName.yaml", packageMap);
+	saveToFile("displayNameToPackageName.yaml", invertMap(packageMap));
+}
+
+const saveToFile = (filename, data) =>
+	fs.writeFileSync(path.join(__dirname, "..", "data", filename), yaml.dump(data), "utf8");
+
+const invertMap = (obj) => Object.fromEntries(Object.entries(obj).map(([k, v]) => [v, k]));
```

Comment:
I don't think this is safe. DisplayName_s are not guaranteed to be unique within a package. For example, it's completely allowed (and somewhat common) to export both a variable and an associated type with the same name. We'll end up clobbering some of the entries here if we do this.

Reply:
This will also fail if multiple packages have exports with the same name.

Reply:
As we discussed, we came to the conclusion to just error on docs build if there exists more than one unscoped package name to display package name and vice versa